lib/reciter.rb

```diff
@@ -1,52 +1,67 @@
-    "This is the rooster that crowed in the morn that woke the priest all shaven and shorn that married the man all tattered and torn that kissed the maiden all forlorn that milked the cow with the crumpled horn that tossed the dog that worried the cat that killed the rat that ate the malt that lay in the house that Jack built.\n" +
-    "This is the farmer sowing his corn that kept the rooster that crowed in the morn that woke the priest all shaven and shorn that married the man all tattered and torn that kissed the maiden all forlorn that milked the cow with the crumpled horn that tossed the dog that worried the cat that killed the rat that ate the malt that lay in the house that Jack built.\n" +
-    "This is the horse and the hound and the horn that belonged to the farmer sowing his corn that kept the rooster that crowed in the morn that woke the priest all shaven and shorn that married the man all tattered and torn that kissed the maiden all forlorn that milked the cow with the crumpled horn that tossed the dog that worried the cat that killed the rat that ate the malt that lay in the house that Jack built.\n"
+    index = 0
+    continous_phrase = ""
+    rhyme = ""
+
+    while index < 12
+      continous_phrase = phrases[index].strip + " " + continous_phrase
+      rhyme += "This is " + continous_phrase.strip + ".\n"
+      index += 1
+    end
+
+    rhyme
   end
 
-  def recite_random_rhyme(random_indices=nil)
-    random_indices ||= generate_random_indices
-
-    phrases = [
-      "the house that Jack built",
-      "the malt that lay in",
-      "the rat that ate",
-      "the cat that killed",
-      "the dog that worried",
-      "the cow with the crumpled horn that tossed",
-      "the maiden all forlorn that milked",
-      "the man all tattered and torn that kissed",
-      "the priest all shaven and shorn that married",
-      "the rooster that crowed in the morn that woke",
-      "the farmer sowing his corn that kept",
-      "the horse and the hound and the horn that belonged to",
-    ]
+  def recite_random_rhyme(random_indices=generate_random_indices)
     continous_phrase = ""
-    random_rhyme = ""
+    rhyme = ""
 
-    for random_index in random_indices
-      continous_phrase = phrases[random_index].strip + " " + continous_phrase
-      random_rhyme += "This is " + continous_phrase.strip + ".\n"
+    for index in random_indices
+      continous_phrase = phrases[index].strip + " " + continous_phrase
+      rhyme += "This is " + continous_phrase.strip + ".\n"
     end
 
-    random_rhyme
+    rhyme
+  end
+
+  def recite_semirandom_rhyme(semirandom_indices=generate_random_indices)
+    # make sure that the phrase 'the house that jack built' is always first
+    semirandom_indices.delete(0)
+    semirandom_indices.unshift(0)
+    continous_phrase = ""
+    rhyme = ""
+
+    for index in semirandom_indices
```

Comment:
Suggestion: This piece of code seems really similar to the code in `recite_random_rhyme` - I wonder if there's some way to extract this responsibility? I also wonder if some sort of nice refactors could be done if `Reciter` just accepted the phrases it should combine, or maybe if a different class did?